go/nbs/root_tracker_test.go

```diff
@@ -112,28 +137,75 @@ func assertDataInStore(slices [][]byte, store chunks.ChunkSource, assert *assert
 	}
 }
 
-func TestChunkStoreManifestFirstWriteByOtherProcess(t *testing.T) {
-	assert := assert.New(t)
-	dir, err := ioutil.TempDir(os.TempDir(), "")
-	assert.NoError(err)
-	defer os.RemoveAll(dir)
+type fakeManifest struct {
```

Comment:
The behavior of fakeManifest is intended to be the same as fileManifest, so you can reason about it as though it were a fileManifest. I've added some big comments to the methods of the `manifest` interface to explain the contract there. This is intended to be a fake that fulfills that contract.